src/source/ObjcGenerator.scala

```diff
@@ -178,7 +178,7 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
     val self = marshal.typename(objcName, r)
 
     refs.header.add("#import <Foundation/Foundation.h>")
-    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) "../" else "") + marshal.headerName(ident)))
+    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) spec.objcExtendedRecordIncludePrefix else "") + marshal.headerName(ident)))
```

Comment:
The diff isn't where the "../" was introduced.  It just included it some of the new/changed lines.  It was already there since the introduction of extended records, which I believe was before the first commit on github.  As Miro points out, the existing include prefix args are for different code, and thus distinct from the new arg you're adding.  But since the "../" was always applied after one of the "find the generated code" prefixes, that's where your new arg is going.  That's workable for some use cases, but not all.

I'm going to put some comments from deeper analysis at the top-level of this PR, rather than on this line.  We can continue the discussion there.